File: android-app/app/src/main/java/com/example/mobileapp/ui/LoginActivity.java
package com.example.mobileapp.ui;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import com.example.mobileapp.databinding.ActivityLoginBinding;
import com.example.mobileapp.models.LoginData;
import com.example.mobileapp.models.LoginRequest;
import com.example.mobileapp.models.LoginResponse;
import com.example.mobileapp.services.ApiClient;
import com.example.mobileapp.utils.SecureStorageManager;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class LoginActivity extends AppCompatActivity {
    
    private ActivityLoginBinding binding;
    private SecureStorageManager secureStorage;
    
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityLoginBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        
        secureStorage = SecureStorageManager.getInstance(this);
        
        // すでにログイン済みの場合
        if (secureStorage.hasToken()) {
            navigateToMain();
            return;
        }
        
        setupViews();
    }
    
    private void setupViews() {
        binding.btnLogin.setOnClickListener(v -> {
            String loginId = binding.etLoginId.getText().toString().trim();
            String password = binding.etPassword.getText().toString().trim();
            
            if (loginId.isEmpty()) {
                binding.tilLoginId.setError("ユーザーIDを入力してください");
            } else if (password.isEmpty()) {
                binding.tilPassword.setError("パスワードを入力してください");
            } else {
                binding.tilLoginId.setError(null);
                binding.tilPassword.setError(null);
                performLogin(loginId, password);
            }
        });
    }
    
    private void performLogin(String loginId, String password) {
        setLoading(true);
        
        LoginRequest request = new LoginRequest(loginId, password);
        ApiClient.getApiService().login(request).enqueue(new Callback<LoginResponse>() {
            @Override
            public void onResponse(Call<LoginResponse> call, Response<LoginResponse> response) {
                setLoading(false);
                
                if (response.isSuccessful() && response.body() != null) {
                    LoginData loginData = response.body().getData();
                    if (loginData != null) {
                        // トークンを保存
                        secureStorage.saveToken(loginData.getToken());
                        
                        Toast.makeText(
                            LoginActivity.this,
                            "ログインしました",
                            Toast.LENGTH_SHORT
                        ).show();
                        
                        navigateToMain();
                    } else {
                        showError("ログインに失敗しました");
                    }
                } else {
                    if (response.code() == 401) {
                        showError("ユーザーIDまたはパスワードが正しくありません");
                    } else {
                        showError("ログインに失敗しました (" + response.code() + ")");
                    }
                }
            }
            
            @Override
            public void onFailure(Call<LoginResponse> call, Throwable t) {
                setLoading(false);
                t.printStackTrace();
                showError("ネットワークエラーが発生しました");
            }
        });
    }
    
    private void setLoading(boolean isLoading) {
        binding.progressBar.setVisibility(isLoading ? View.VISIBLE : View.GONE);
        binding.btnLogin.setEnabled(!isLoading);
        binding.etLoginId.setEnabled(!isLoading);
        binding.etPassword.setEnabled(!isLoading);
    }
    
    private void showError(String message) {
        binding.tvError.setText(message);
        binding.tvError.setVisibility(View.VISIBLE);
    }
    
    private void navigateToMain() {
        Intent intent = new Intent(this, MainActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
        startActivity(intent);
        finish();
    }
}

File: android-app/app/src/main/java/com/example/mobileapp/ui/LoginActivity.kt
package com.example.mobileapp.ui

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import com.example.mobileapp.databinding.ActivityLoginBinding
import com.example.mobileapp.models.LoginRequest
import com.example.mobileapp.services.ApiClient
import com.example.mobileapp.utils.SecureStorageManager
import kotlinx.coroutines.launch

class LoginActivity : AppCompatActivity() {
    
    private lateinit var binding: ActivityLoginBinding
    private lateinit var secureStorage: SecureStorageManager
    
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityLoginBinding.inflate(layoutInflater)
        setContentView(binding.root)
        
        secureStorage = SecureStorageManager.getInstance(this)
        
        // すでにログイン済みの場合
        if (secureStorage.hasToken()) {
            navigateToMain()
            return
        }
        
        setupViews()
    }
    
    private fun setupViews() {
        binding.btnLogin.setOnClickListener {
            val loginId = binding.etLoginId.text.toString().trim()
            val password = binding.etPassword.text.toString().trim()
            
            when {
                loginId.isEmpty() -> {
                    binding.tilLoginId.error = "ユーザーIDを入力してください"
                }
                password.isEmpty() -> {
                    binding.tilPassword.error = "パスワードを入力してください"
                }
                else -> {
                    binding.tilLoginId.error = null
                    binding.tilPassword.error = null
                    performLogin(loginId, password)
                }
            }
        }
    }
    
    private fun performLogin(loginId: String, password: String) {
        setLoading(true)
        
        lifecycleScope.launch {
            try {
                val request = LoginRequest(loginId, password)
                val response = ApiClient.getApiService().login(request)
                
                if (response.isSuccessful) {
                    val loginData = response.body()?.data
                    if (loginData != null) {
                        // トークンを保存
                        secureStorage.saveToken(loginData.token)
                        
                        Toast.makeText(
                            this@LoginActivity,
                            "ログインしました",
                            Toast.LENGTH_SHORT
                        ).show()
                        
                        navigateToMain()
                    } else {
                        showError("ログインに失敗しました")
                    }
                } else {
                    when (response.code()) {
                        401 -> showError("ユーザーIDまたはパスワードが正しくありません")
                        else -> showError("ログインに失敗しました (${response.code()})")
                    }
                }
            } catch (e: Exception) {
                e.printStackTrace()
                showError("ネットワークエラーが発生しました")
            } finally {
                setLoading(false)
            }
        }
    }
    
    private fun setLoading(isLoading: Boolean) {
        binding.progressBar.visibility = if (isLoading) View.VISIBLE else View.GONE
        binding.btnLogin.isEnabled = !isLoading
        binding.etLoginId.isEnabled = !isLoading
        binding.etPassword.isEnabled = !isLoading
    }
    
    private fun showError(message: String) {
        binding.tvError.text = message
        binding.tvError.visibility = View.VISIBLE
    }
    
    private fun navigateToMain() {
        val intent = Intent(this, MainActivity::class.java)
        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
        startActivity(intent)
        finish()
    }
}
